test/jdk/java/nio/channels/FileChannel/LargeMapTest.java

```diff
@@ -38,42 +41,44 @@
  * @summary Ensure that memory mapping beyond 32-bit range does not cause an
  *          EXCEPTION_ACCESS_VIOLATION.
  * @requires vm.bits == 64
- * @run main/othervm/timeout=240 LargeMapTest
+ * @run main/othervm LargeMapTest
  */
 public class LargeMapTest {
-    private static final String FILE = "test.dat";
-    private static final long LENGTH = 8000000000L;
-    private static final long OFFSET = 3704800000L;
-    private static final int  BUFSIZ = 100000;
+    private static final long LENGTH = (1L << 32) + 512;
+    private static final int  EXTRA  = 1024;
+    private static final long BASE   = LENGTH - EXTRA;
 
     public static void main(String[] args) throws IOException {
         System.out.println(System.getProperty("sun.arch.data.model"));
         System.out.println(System.getProperty("os.arch"));
         System.out.println(System.getProperty("java.version"));
 
-        Path p = Path.of(FILE);
+        System.out.println("  Writing large file...");
+        long t0 = System.nanoTime();
+        Path p = Files.createTempFile("test", ".dat");
+        Files.delete(p); // re-create as sparse
         p.toFile().deleteOnExit();
-        try (FileChannel fc = FileChannel.open(p, CREATE, WRITE)) {
-            fc.position(LENGTH - 1);
-            fc.write(ByteBuffer.wrap(new byte[] {27}));
+        ByteBuffer bb;
+        try (FileChannel fc = FileChannel.open(p, CREATE_NEW, SPARSE, WRITE)) {
+            fc.position(BASE);
+            Random r = new Random(System.nanoTime());
```

Comment:
It might not matter here but we usually use RandomFactory to allow for reproducibility in the event of test failure.

Reply:
Thanks for the hint! It shouldn't matter here, but I'll change it.